### ISSUE TITLE
Query modifier UI

### ISSUE THREAD

**Issue by jboning** [livegrep/livegrep#74]
Inspired by #71, filing a separate issue to discuss query modifier UI in general.

The query parser currently understands these tags:
 - `file:`
 - `repo:`
 - `tags:`
 - `case:`
 - `lit:`
 - `max_matches:`

I'd group these as follows:
 - The `file:` and `repo:` tags are like extra search terms, filtering query results. (`tags:` seems also to fall into this category, but additionally slightly modifies treatment of the main content query?)
 - `case:` and `lit:` are modifiers for the meaning of the main query.
 - `max_matches:` is a modifier on the overall search behaviour.

For the second group (`case` and `lit`), there are a few rough edges:
 - The default behaviour/setting is not obvious (though it's documented on the help page).
 - If a user typically uses a particular mode, the setting isn't sticky.
 - It's not clear how these modifiers interact with `file` and `repo` terms--whether you can ask for `file:lit:foo.c`. Since you can't, there's no way to control the case-sensitivity of a `file` field (after #72).

---

One possibility is to move the query modifiers into separate UI controls, something like this:

    Case:  (x) auto   ( ) match   ( ) ignore
    Query: (x) regex  ( ) literal

This:
 - makes the query interpretation explicit in the UI (especially the "auto" case-sensitivity detection, which I didn't even realize existed!)
 - lets us store the user's case/query preferences in a cookie and apply them again at next page load

With this UI, I think it would be least surprising to apply the settings to both the main query and `file:` terms, which would be a partial reversal of #72.

**Comment by tpcwang** [comment]
I'm not sure what the UI should look like, but making the operators more easily discoverable seems good to me. I was originally thinking about a drop down menu like what [DXR](https://dxr.mozilla.org/mozilla-central/source/) does, but I can see separate controls working as well.

I wouldn't mind a partial reversal of #72 as long as the behavior is intuitive to the user. Thanks for sharing your idea.

**Comment by jboning** [comment]
Hearing no wailing or gnashing of teeth at that idea, I'm going to go ahead and build it!
